### DISCUSSION TITLE
dependencies for distributable

### DISCUSSION THREAD

**Discussion by garrilla** [nick-thompson/react-juce#271] [Q&A]
If i use Blueprint, what dependencies does a distributable plugin require?

For example, would an end-user also need to load Node/NPM?

**Comment by nick-thompson** [comment]
Good question! The only explicit javascript dependencies are listed here: https://github.com/nick-thompson/react-juce/blob/master/packages/react-juce/package.json#L40, but each one is specifically tested such that it can be bundled (via webpack, parcel, rollup, etc) with your application code into a single javascript file. That single javascript file then completely represents your app's UI and you can embed it into your JUCE application (i.e. with BinaryData) and load it from memory, or fetch it from a server.

So, to your specific question, Node and npm are only used here for dependency management + builds. Once you have bundled your app, node and npm are no longer involved and your end users never need to think about them!
